packages/flutter/test/cupertino/search_field_test.dart

```diff
@@ -0,0 +1,475 @@
+        home: Center(
+          child: CupertinoSearchTextField(),
+        ),
+      ),
+    );
+
+    Text placeholder = tester.widget(find.text('Search'));
+    expect(placeholder.style!.color!.value,
+        CupertinoColors.secondaryLabel.darkColor.value);
+
+    await tester.pumpAndSettle();
+
+    await tester.pumpWidget(
+      const CupertinoApp(
+        theme: CupertinoThemeData(brightness: Brightness.light),
+        home: Center(
+          child: CupertinoSearchTextField(),
+        ),
+      ),
+    );
+
+    placeholder = tester.widget(find.text('Search'));
+    expect(placeholder.style!.color!.value,
+        CupertinoColors.secondaryLabel.color.value);
+  });
+
+  testWidgets(
+    "placeholderStyle modifies placeholder's style and doesn't affect text's style",
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        const CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              placeholder: 'placeholder',
+              style: TextStyle(
+                color: Color(0x00FFFFFF),
+                fontWeight: FontWeight.w300,
+              ),
+              placeholderStyle: TextStyle(
+                color: Color(0xAAFFFFFF),
+                fontWeight: FontWeight.w600,
+              ),
+            ),
+          ),
+        ),
+      );
+
+      final Text placeholder = tester.widget(find.text('placeholder'));
+      expect(placeholder.style!.color, const Color(0xAAFFFFFF));
+      expect(placeholder.style!.fontWeight, FontWeight.w600);
+
+      await tester.enterText(find.byType(CupertinoSearchTextField), 'input');
+      await tester.pump();
+
+      final EditableText inputText = tester.widget(find.text('input'));
+      expect(inputText.style.color, const Color(0x00FFFFFF));
+      expect(inputText.style.fontWeight, FontWeight.w300);
+    },
+  );
+
+  testWidgets(
+    'prefix widget is in front of the text',
+    (WidgetTester tester) async {
+      final FocusNode focusNode = FocusNode();
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              focusNode: focusNode,
+              controller: TextEditingController(text: 'input'),
+            ),
+          ),
+        ),
+      );
+
+      expect(
+        tester.getTopRight(find.byIcon(CupertinoIcons.search)).dx + 3.8,
+        tester.getTopLeft(find.byType(EditableText)).dx,
+      );
+
+      expect(
+        tester.getTopLeft(find.byType(EditableText)).dx,
+        tester.getTopLeft(find.byType(CupertinoSearchTextField)).dx +
+            tester.getSize(find.byIcon(CupertinoIcons.search)).width +
+            9.8,
+      );
+    },
+  );
+
+  testWidgets(
+    'suffix widget is after the text',
+    (WidgetTester tester) async {
+      await tester.pumpWidget(
+        CupertinoApp(
+          home: Center(
+            child: CupertinoSearchTextField(
+              controller: TextEditingController(text: 'Hi'),
```

Comment:
same

Reply:
controller needs to have a text in order for the editable to show up